azurerm/internal/services/databricks/databricks_workspace_resource.go

```diff
@@ -85,25 +85,76 @@ func resourceDatabricksWorkspace() *pluginsdk.Resource {
 				Default:  false,
 			},
 
+			"public_network_access_enabled": {
+				Type:     pluginsdk.TypeBool,
+				ForceNew: true,
+				Optional: true,
+				Default:  true,
+			},
+
+			"require_network_security_group_rules": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(databricks.RequiredNsgRulesAllRules),
+					string(databricks.RequiredNsgRulesNoAzureDatabricksRules),
+					string(databricks.RequiredNsgRulesNoAzureServiceRules),
+				}, false),
+			},
+
 			"custom_parameters": {
 				Type:     pluginsdk.TypeList,
 				Optional: true,
 				Computed: true,
 				MaxItems: 1,
 				Elem: &pluginsdk.Resource{
 					Schema: map[string]*pluginsdk.Schema{
+						"load_balancer_backend_pool_name": {
+							Type:         pluginsdk.TypeString,
+							ForceNew:     true,
+							Optional:     true,
+							AtLeastOneOf: workspaceCustomParametersString(),
+						},
+
+						"load_balancer_id": {
```

Comment:
can you get the name from the id?

Reply:
No, I cannot get the backend name from the load balancer but I can get the load balancer ID from the backend pool id. 🤔

Reply:
Fixed... I have removed the two custom parameters and replaced it with a single top level parameter.